ZigbeeNet/CC/CCZnp.cs

```diff
@@ -370,68 +272,247 @@ public void SendAREQ(ZpiObject zpiObject)
                 return;
             }
 
-            if ((zpiObject.SubSystem == SubSystem.SYS && zpiObject.CommandId == 0) 
-                || (zpiObject.SubSystem == SubSystem.SAPI && zpiObject.CommandId == 9)) //resetReq or systemReset
+            if (zpiObject.SubSystem == SubSystem.SYS && zpiObject.CommandId == 0
+                || zpiObject.SubSystem == SubSystem.SAPI && zpiObject.CommandId == 9) //resetReq or systemReset
             {
                 _resetting = true;
 
                 // clear all pending requests, since the system is reset
                 _requestQueue = new ConcurrentQueue<ZpiObject>();
 
-                this.AsyncResponse += (object sender, ZpiObject e) =>
+                AsyncResponse += (sender, e) =>
                 {
-                    if (e.Type == MessageType.AREQ && e.SubSystem == SubSystem.SYS && e.CommandId == (byte)SYS.SysCommand.resetInd)
-                    {
-                        _resetting = false;
-                    }
+                    if (e.Type == MessageType.AREQ && e.SubSystem == SubSystem.SYS &&
+                        e.CommandId == (byte) SysCommand.resetInd) _resetting = false;
                 };
             }
 
-            Unpi.Send((int)MessageType.AREQ, (int)zpiObject.SubSystem, zpiObject.CommandId, zpiObject.Frame);
+            unpi.Send((int) MessageType.AREQ, (int) zpiObject.SubSystem, zpiObject.CommandId, zpiObject.Frame);
         }
 
         private void ScheduleNextSend()
         {
             _sreqRunning = null;
 
-            ZpiObject next = new ZpiObject();
-            if (_requestQueue.TryDequeue(out next)) { 
-                Request(next);
-            }
+            var next = new ZpiObject();
+            if (_requestQueue.TryDequeue(out next)) Request(next);
         }
 
         private void ParseIncomingData(ZpiObject request, byte[] buffer)
         {
-            if (buffer == null || buffer.Length == 0)
-            {
-                throw new NullReferenceException("Buffer is empty");
-            }
+            if (buffer == null || buffer.Length == 0) throw new NullReferenceException("Buffer is empty");
```

Comment:
We should avoid throwing NullReferenceException manually like mentioned by @nicolaiw in Issue #15

Reply:
Not my code.

Reply:
Oh i thought i fixed it already. Sorry